sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,338 @@
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) Codeplay. All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 10 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for GPU devices and
+only when using the Level Zero backend.
+
+== Overview
+
+This extension provides new APIs for allocating and deallocating exportable
+device memory in SYCL, and obtaining a handle to that memory which can be used
+in external APIs. This is useful when applications want to share device memory
+with other third-party APIs.
+
+Without the ability to allocate exportable memory and obtain an interoperable
+handle, applications would have to copy device memory allocated by one API to
+the host, then copy that host memory back to the device in a memory region
+allocated by a second API. If the second API modifies that memory, then this
+process would have to be repeated in the opposite direction in order for the
+first API to see the changes made to that memory.
+
+This extension enables copy-free sharing of SYCL allocated device memory with
+external APIs.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_MEMORY_EXPORT` to one of the values defined in the
+table below. Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Rev | Description
+|1   | Initial draft of the proposal
+|======================
+
+=== Querying device support
+
+We provide the following device aspect to query for support of exporting memory.
+
+[frame="none",options="header"]
+|======================
+|Device descriptor |Description
+|`aspect::ext_oneapi_memory_export_linear` | Indicates if the device supports
```

Comment:
How about `aspect::ext_oneapi_exportable_device_mem`, which matches your other names `alloc_exportable_device_mem` and `free_exportable_mem`.  If you add support for exportable optimized memory in the future, you can name it something like `aspect::ext_oneapi_exportable_device_mem_optimized`.